### PR TITLE
services/horizon: Display correct Transaction.FeePaid value

### LINKED ISSUE #1350: horizon reports incorrect fee if actual fee is below the max fee selected (with core v11)
When trying the new fee logic (protocol v11) on testnet, we noticed that horizon shows a confusing value of "fee_paid".

Let's say, you set a fee to 10000 (instead of min fee of 100) to make sure that the payment would go through even if the network is busy.

However, test net is currently not congested. So we expect the min fee of 100 to be used.

After submitting the transaction, and looking at horizon response - we see that **fee_paid** is set to 10000 (as would happen on protocol v10), while the real balance gets decreased by 100.

So you actually pay a min fee, but horizon reports that a fee of 10000 was paid.

------------------

Here's the example of tx, which shows the "fee_paid": 10000:

https://horizon-testnet.stellar.org/transactions/99052ed553e08c05bb4b5f230ece6b250fad63d9a91955254996d56520056556/

Here's the related result XDR parsed:

https://www.stellar.org/laboratory/#xdr-viewer?input=AAAAAAAAAGQAAAAAAAAAAQAAAAAAAAABAAAAAAAAAAA%3D&type=TransactionResult&network=public

As you can see the actual fee paid is 100. That's what horizon should show as well.

### PR DESCRIPTION
**Pull request by bartekn**
## Summary

The changes implemented in [CAP-0005](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0005.md) were not reflected in Horizon. `Transaction.FeePaid` was displaying the maximum fee bid instead of the fee that was actually paid.

This is a bug fix so should be released in a patch release (0.17.7). The new fields will be added in 0.18.0 and a breaking field name change will be updated in 0.19.0.

Close #1350.

### Summary of changes

* Created a new `fee_charged` column in `history_transactions` table that represents the actual fee paid. If the column value is `NULL` (before DB reingestion) the `fee_paid` value will be returned instead of `0`.
* The old field `fee_paid` name was not changed because it would require full table rewrite. The name was actually changed to `MaxFee` in Horizon code.
* A new field `max_fee` was added to Transaction resource in `protocols/horizon`. It should activate in 0.18.0.
* The `fee_paid` in Transaction resource in `protocols/horizon` was deprecated. It should be changed to `fee_charged` in 0.18.0.

### Known limitations & issues

* The `/fee_stats` endpoint will display the stats for the bids instead of actual fees paid after this PR is merged.
* ~Tests are failing because of test scenarios that are currently being rebuilt.~

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.

### CODE REVIEW

File: services/horizon/internal/db2/schema/migrations/17_transaction_fee_paid.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- +migrate Up
+
+ALTER TABLE history_transactions ADD fee_charged integer;
```
Comment (tomquisel) [inline services/horizon/internal/db2/schema/migrations/17_transaction_fee_paid.sql]: I'd suggest adding `fee_charged` and then also renaming `fee_paid` to `max_fee`.

File: protocols/horizon/main.go [hunk 2]
```diff
@@ -401,17 +401,22 @@ type Transaction struct {
...
-	ResultMetaXdr   string    `json:"result_meta_xdr"`
-	FeeMetaXdr      string    `json:"fee_meta_xdr"`
-	MemoType        string    `json:"memo_type"`
-	Memo            string    `json:"memo,omitempty"`
-	Signatures      []string  `json:"signatures"`
-	ValidAfter      string    `json:"valid_after,omitempty"`
-	ValidBefore     string    `json:"valid_before,omitempty"`
+	// Deprecated - remove in: horizon-v0.19.0
+	// Change the name to `fee_charged`
+	FeeCharged int32 `json:"fee_paid"`
```
[the remark this answers is not in the document]
Comment (bartekn) [reply protocols/horizon/main.go]: Yes, this was fine for Horizon (response field name was still `fee_paid`) but I forgot that `clients/horizonclient` package is using `protocols/horizon` now. So I left `FeePaid int32 `json:"fee_paid"`` as it was, I added `FeeCharged` and `MaxFee` that are currently hidden in Horizon responses (to not add new fields in a patch release). We need to coordinate with @poliha to release `horizonclient` with removed field.

File: services/horizon/internal/db2/history/main.go [hunk 3]
```diff
@@ -300,25 +300,28 @@ type TradesQ struct {
...
-	UpdatedAt        time.Time   `db:"updated_at"`
+	TransactionHash  string    `db:"transaction_hash"`
+	LedgerSequence   int32     `db:"ledger_sequence"`
+	LedgerCloseTime  time.Time `db:"ledger_close_time"`
+	ApplicationOrder int32     `db:"application_order"`
+	Account          string    `db:"account"`
+	AccountSequence  string    `db:"account_sequence"`
+	// FeePaid defines the max fee set by the user for this transaction.
+	// The column name was not changed because of the painful DB migration.
+	MaxFee          int32       `db:"fee_paid"`
```
[the remark this answers is not in the document]
Comment (bartekn) [reply services/horizon/internal/db2/history/main.go]: I didn't know that `REMOVE COLUMN` was fast. Changed the migration code.

**Comment by bartekn** [comment]
@tomquisel thanks for a good feedback. Let's wait for coordination with `horizonclient` maintainers. I'm not sure if removing or adding a a field in `protocols/horizon` is a breaking change in the context of that package but it probably is so we need to release a new `horizonclient` with new/removed fields.

**Comment by poliha** [comment]
Yes removing/renaming `FeePaid`  will result in a breaking change in `horizonclient` because this field is expected when querying for transactions. Changes to these fields will require a release.

**Pull request merged**